### PR TITLE
Added tests for verified and unverified login attempts

### PR DESCRIPTION
**Pull request by julen**
Tests that user will be able to either
* log in and therefore be redirected to `settings.LOGIN_REDIRECT_URL` (optional verification), or
* reach the email verification view (mandatory verification)

Also removed the empty `test_login_view()` test.

### CODE REVIEW

File: allauth/account/tests.py [hunk 1]
```diff
@@ -301,27 +330,6 @@ def test_ajax_login_success(self):
...
-        user.set_password('doe')
-        user.save()
-        EmailAddress.objects.create(user=user,
-                                    email='john@example.com',
-                                    primary=True,
-                                    verified=False)
-        resp = self.client.post(reverse('account_login'),
-                                {'login': 'john',
-                                 'password': 'doe'},
-                                HTTP_X_REQUESTED_WITH='XMLHttpRequest')
```
Comment (pennersr) [inline allauth/account/tests.py]: Why did you remove this test case, as it tests AJAX based logins, which is not tested by the tests you added?

File: allauth/account/tests.py [hunk 2]
```diff
@@ -301,27 +330,6 @@ def test_ajax_login_success(self):
         data = json.loads(resp.content.decode('utf8'))
         self.assertEqual(data['location'], '/accounts/profile/')
 
-    @override_settings(
-        ACCOUNT_EMAIL_VERIFICATION=app_settings.EmailVerificationMethod
-        .MANDATORY,
-        ACCOUNT_AUTHENTICATION_METHOD=app_settings.AuthenticationMethod
```
Comment (pennersr) [inline allauth/account/tests.py]: Please override the `ACCOUNT_AUTHENTICATION_METHOD` method as well, as you are logging in by username.

File: allauth/account/tests.py (hunk 1 above)
Comment (julen) [inline allauth/account/tests.py]: It's exactly the same test as `test_login_unverified_account_mandatory`. The fact that a view is requested with the `X-Requested-With: XMLHttpRequest` header shouldn't be tested for every single scenario IMO.

File: allauth/account/tests.py (hunk 2 above)
Comment (julen) [inline allauth/account/tests.py]: Unless I'm missing something, `AccountTests` already defaults to `AuthenticationMethod.USERNAME`, and [the `@override_settings` decorator](https://docs.djangoproject.com/en/1.7/topics/testing/tools/#django.test.override_settings) overrides settings only for a method. So is it really necessary?
